src/pages/en/guides/content-collections.md

```diff
@@ -0,0 +1,475 @@
+- `getEntry` — get a specific entry in a collection by file name
+
+These functions will be typed based on collections that exist. For example, `getCollection('banana')` will raise a type error if there is no `src/content/banana/`.
+
+```astro
+---
+import { getCollection, getEntry } from 'astro:content';
+// Get all `src/content/blog/` entries
+const allBlogPosts = await getCollection('blog');
+// Filter blog posts by entry properties
+const draftBlogPosts = await getCollection('blog', ({ id, slug, data }) => {
+  return data.status === 'draft';
+});
+// Get a specific blog post by file name
+const enterprise = await getEntry('blog', 'enterprise.md');
+---
+```
+
+### Return type
+
+Assume the `blog` collection schema looks like this:
+
+```ts
+// src/content/config.ts
+import { defineCollection, z } from 'astro:content';
+
+const blog = defineCollection({
+  schema: {
+    title: z.string(),
+    slug: z.string(),
+    image: z.string().optional(),
+    tags: z.array(z.string()),
+  },
+});
+
+export const collections = { blog };
+```
+
+`await getCollection('blog')` will return entries of the following type, with the type of `data` inferred from your schema:
+
+```ts
+{
+  // Unique ID using file path relative to src/content/[collection]
+  // Ex. for content/[collection]/file-1.md...
+  id: 'file-1.md' | 'file-2.md' | ...;
+	// URL-ready slug using ID with file extension stripped
+  // Ex. for content/[collection]/file-1.md...
+	slug: 'file-1' | 'file-2' | ...;
+  // Inferred from collection schema
+  // Defaults to `any` if no schema is configured
+  data: {
+    title: string;
+    slug: string;
+    image?: string;
+    tags: string[];
+  };
+  // Raw body of the Markdown or MDX document
+  body: string;
+}
+```
+
+:::note
+The `body` is the *raw* content of the file. This ensures builds remain performant by avoiding expensive rendering pipelines. See [**Rendering content**](#rendering-content-with-renderentry) to compile this body to a `Content` component for use in your Astro files.
+:::
+
+### Getting from nested directories
+
+When getting a collection that includes files in nested directories, the result will be a flat array of entries. The nested directory structure will be reflected in each entry’s `id`.
+
+```ts
+const docsEntries = await getCollection('docs');
+console.log(docsEntries)
+/*
+[
+  { id: 'en/getting-started.md', slug: 'en/getting-started', data: {...} },
+  { id: 'en/structure.md', slug: 'en/structure', data: {...} },
+  { id: 'es/getting-started.md', slug: 'es/getting-started', data: {...} },
+  { id: 'es/structure.md', slug: 'es/structure', data: {...} },
+  ...
+]
+*/
+```
+
+To fetch from a subdirectory within a collection, you can use `getCollection` with a filter:
+
+```astro
+---
+import { getCollection } from 'astro:content';
+
+const enDocs = await getCollection('docs', ({ id }) => {
+  // Where `id` is 'en/page-1.md' | 'en/page-2.md' | ...
+  return id.startsWith('en/');
+});
+---
+```
```

Comment:
```suggestion
### Data returned from a collections query

`getCollection()` and `getEntry()` will return an array of entries (or a single entry) that include:

 -  `id` - a unique ID using the file path relative to `src/content/[collection]`.
 - `slug` - the URL-ready slug using ID without file extension.
 - `data` - an object of your frontmatter properties inferred from the collection schema. (Defaults to `any` if no schema is configured.)
 - `body` - a string containing the raw body of the Markdown or MDX document.

```ts
const docsEntries = await getCollection('docs');
console.log(docsEntries)
/*
[
  { id: 'columbia.md', slug: 'columbia', data: {...}, body: '...' },
  { id: 'endeavour.md', slug: 'endeavour', data: {...}, body: '...'  },
  { id: 'enterprise.md', slug: 'enterprise', data: {...}, body: '...' }
]
*/
```
:::note
The `body` is the *raw* content of the file. This ensures builds remain performant by avoiding expensive rendering pipelines. See [**Rendering content**](#rendering-content-with-renderentry) to compile this body to a `<Content />` component for use in your Astro files.
:::

Add a filtering function as a second parameter to `getCollections()` allows you to return a subset of a collection's entries. For example, to fetch an entire subdirectory within a collection:

```astro
---
import { getCollection } from 'astro:content';

const enDocs = await getCollection('docs', ({ id }) => {
  return id.startsWith('en/');
});
---
```
```ts
console.log(enDocs)
/*
[
  { id: 'en/getting-started.md', slug: 'en/getting-started', data: {...}, body: '...'  },
  { id: 'en/structure.md', slug: 'en/structure', data: {...}, body: '...'  },
  { id: 'es/getting-started.md', slug: 'es/getting-started', data: {...}, body: '...'  },
  { id: 'es/structure.md', slug: 'es/structure', data: {...}, body: '...'  },
  ...
]
*/
```
OK, this got long and icky, so I hope the formatting makes sense. But, I was thining of something like this to replace the "return value" that's going to end up in API reference, and combining it with the following section, which I'm not sure needs to be its own section? It's just another example of filtering, albeit a handy example, so included here under querying is fine I think.

``````

Reply:
I love everything about this ❤️